File: examples/mortar-app/src/main/java/architect/examples/mortar_app/screen/home/HomePresenter.java
package architect.examples.mortar_app.screen.home;

import android.os.Bundle;

import mortar.ViewPresenter;

/**
 * @author Lukasz Piliszczuk - lukasz.pili@gmail.com
 */
public class HomePresenter extends ViewPresenter<HomeView> {

//    private static int count = 0;
//
//    //    @NavigationParam
//    private final String name;
//
//    //    @NavigationResult
//    private final String result;
//
//    //    @ScreenData
//    private final HomeState state;

//    public HomePresenter(HomeState state, String name, String result) {
//        this.state = state;
//        this.name = name;
//        this.result = result;
//
//        if (result != null) {
//            Timber.d("Home presenter with result: %s", result);
//        }
//    }

    @Override
    protected void onLoad(Bundle savedInstanceState) {
//        Timber.d("Home onLoad with random = %d", state.random);
//
//        getView().titleTextView.setText(name);
//        getView().subtitleTextView.setText("Random number: " + state.random);
    }

//    @Parcel(parcelsIndex = false)
//    public static class HomeState {
//
//        int random;
//
//        public HomeState() {
//            this.random = new Random().nextInt(100);
//        }
//
//        @ParcelConstructor
//        public HomeState(int random) {
//            this.random = random;
//        }
//    }
}

File: kotlin_app/src/main/kotlin/architect/kotlinapp/mvp/home/HomePresenter.kt
package architect.kotlinapp.mvp.home

import architect.kotlinapp.MainActivity
import architect.kotlinapp.mvp.home.content.HomeContentPresenter
import architect.kotlinapp.mvp.home.drawer.HomeDrawerPresenter
import architect.robot.*
import autodagger.AutoComponent
import mortar.ViewPresenter
import org.parceler.Parcel

/**
 * @author Lukasz Piliszczuk - lukasz.pili@gmail.com
 */
@AutoScreen(
        component = AutoComponent(dependencies = arrayOf(MainActivity::class)),
        pathView = HomeView::class,
        subScreens = arrayOf(
                ContainsSubscreen(type = HomeContentPresenter::class, name = "content"),
                ContainsSubscreen(type = HomeDrawerPresenter::class, name = "drawer")
        )
)
class HomePresenter(@field:NavigationParam val navParam1: String,
                    @field:NavigationParam val navParam2: String,
                    @field:NavigationResult val navResult: String?,
                    @field:ScreenData val state: HomePresenter.State) : ViewPresenter<HomeView>() {

//    @NavigationParam
    private lateinit var yo: String

//    @NavigationParam
    private var foo: String? = null

    @Parcel(parcelsIndex = false)
    class State {

    }
}